### PR TITLE
Feature/plugin updates

### PR DESCRIPTION
**Pull request by nakulgan**

### CODE REVIEW

**Comment by 5c077yP** [comment]
can you also please adjust the readme, it think in the service section now the config files looks different

**Comment by nakulgan** [comment]
@5c077yP @tkuTM did you guys review this

**Comment by tkuTM** [comment]
I am fine..it is mostly the changes i made in my PR..

File: config/services.example.json [hunk 1]
```diff
@@ -34,9 +34,49 @@
         "error_params": {
           "scaling": 2,
           "min_th_up": 0.1,
```
Comment (Puneeth-n) [inline config/services.example.json]: `min_th_up` and `min_th_low` has changed. Needs to be updated.

File: config/services.example.json [hunk 2]
```diff
@@ -34,9 +34,49 @@
...
-          "min_th_low": -0.1,
+          "min_th_low": -0.1
+        }
+      }
+    }
+  },
+  "SeasonalDecompositionEnsemble": {
+    "scheduler_options": {
+      "interval_secs": 30,
+      "plugin": "PollSeasonalDecompositionEnsemble",
```
Comment (5c077yP) [inline config/services.example.json]: i think this is wrong now? it shouldnt it be `SeasonalDecompositionEnsemble`?

Comment (5c077yP) [inline config/services.example.json]: if so we should update this for the other config above as well

File: test/fixtures/config.py [hunk 3]
```diff
@@ -65,10 +65,45 @@
...
+                'interval': 1,
+                'error_params': {}
             }
         }
-    }
+    },
+    'SeasonalDecompositionEnsemble': {
+        'scheduler_options': {
+            'interval_secs': 180,
+            'plugin': 'PollSeasonalDecompositionEnsemble',
```
Comment (5c077yP) [inline test/fixtures/config.py]: also here.. i think it should be `SeasonalDecompositionEnsemble `

File: test/test_scheduler.py [hunk 4]
```diff
@@ -25,11 +23,21 @@ def test_scheduler_celerybeat_schedule_should_be_set(self):
...
             'SeasonalDecomposition': {
                 'task': 'lib.app.task_runner',
                 'schedule': timedelta(seconds=300),
-                'args': (poll_tasks.PollSeasonalDecomposition, {})
+                'args': (PollTask, {'plugin_name': 'PollSeasonalDecomposition'})
+            },
+            'SeasonalDecompositionEnsemble': {
+                'task': 'lib.app.task_runner',
+                'schedule': timedelta(seconds=180),
+                'args': (PollTask, {'plugin_name': 'PollSeasonalDecompositionEnsemble'})
```
Comment (5c077yP) [inline test/test_scheduler.py]: hmm .. how comes that this works?

File: bin/scheduler.py [hunk 5]
```diff
@@ -24,9 +24,8 @@
 
 for algorithm, algo_config in services_config.iteritems():
     scheduler_options = algo_config['scheduler_options']
-    plugin = getattr(poll_tasks, scheduler_options['plugin'])
     CELERYBEAT_SCHEDULE[algorithm] = {
         'task': 'lib.app.task_runner',
         'schedule': timedelta(seconds=scheduler_options['interval_secs']),
-        'args': (plugin, scheduler_options['plugin_args'])
+        'args': (PollTask, {'plugin_name': scheduler_options['plugin']})
```
Comment (5c077yP) [inline bin/scheduler.py]: so we cannot pass the plugin_args anymore ? not that we used them, but if so you can remove it from the README?

Comment (5c077yP) [inline bin/scheduler.py]: also i wonder if `'plugin'` should be the same as `algorithm` here? if so we can use algorithm instead and also remove `'plugin'` from the readme ?

Comment (5c077yP) [inline bin/scheduler.py]: sorry, actually if we leave plugin we could have two times the same algorithm but with different schedules ... so let's leave it, but maybe we default to algorithm if plugin is not given ...

File: config/services.example.json (hunk 1 above)
Comment (nakulgan) [inline config/services.example.json]: Changed to what?

File: config/services.example.json (hunk 2 above)
Comment (nakulgan) [inline config/services.example.json]: It isn't. The plugin is a scheduler option. You could use a PollTask or the plugin itself.

File: test/test_scheduler.py (hunk 4 above)
Comment (nakulgan) [inline test/test_scheduler.py]: Which part

File: config/services.example.json (hunk 2 above)
Comment (5c077yP) [inline config/services.example.json]: but you removed the different PollTasks. There is only PollTask by default and it loads the plugin ... but not the a poll plugin

Comment (nakulgan) [inline config/services.example.json]: Ah, yes. You're right

File: config/services.example.json (hunk 1 above)
Comment (Puneeth-n) [inline config/services.example.json]: `minimal_upper_threshold` and `minimal_lower_threshold` see below the config of ensemble plugin.

Comment (Puneeth-n) [inline config/services.example.json]: Also, the format of the plugin has changed and looks like this:

````
"SeasonalDecomposition": {
    "scheduler_options": {
      "interval_secs": 30,
      "plugin": "SeasonalDecomposition",
      "plugin_args": {}
    },
    "worker_options": {
      "stl_service1": {
        "metric": "system.loadavg",
        "period_length": 6,
        "seasons": 2,
        "interval": 300,
        "error_params": {
          "error_type": "norm",
          "error_handling": "tukey",
          "scaling": 2,
          "minimal_upper_threshold": 0.1,
          "minimal_lower_threshold": -0.1
        }
      }
    }
  }
````

File: config/services.example.json [hunk 6]
```diff
@@ -2,8 +2,9 @@
   "TukeysFilter": {
     "scheduler_options": {
       "interval_secs": 30,
-      "plugin": "PollTukeysFilter",
-      "plugin_args": {}
+      "plugin_args": {
+        "name": "PollTukeysFilter"
```
Comment (tkuTM) [inline config/services.example.json]: no Poll needed anymore

Comment (5c077yP) [inline config/services.example.json]: true, this should be fixed!

**Comment by 5c077yP** [comment]
@tkuTM can you also please update the readme section of `service.json` , there it still references to plugin, but it should be in plugin_args ..

**Comment by 5c077yP** [comment]
:+1:

**Comment by Puneeth-n** [comment]
:+1:

**Pull request merged**